include/zephyr/drivers/pwm/pca9685.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 Nick Ward <nix.ward@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PWM_PCA9685_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PWM_PCA9685_H_
+
+/* See PCA9585 datasheet Rev. 4 - 16 April 2015 section 7.3.5 */
+
+#define PCA9685_OSC_CLOCK_INTERNAL 25000000
+#define PCA9685_OSC_CLOCK_MAX	   50000000
+
+#define PCA9685_CHANNEL_CNT 16
+
+#define PCA9685_PWM_STEPS 4096
+
+#define PCA9685_PRE_SCALE_MIN 0x03
+#define PCA9685_PRE_SCALE_MAX 0xFF
+
+#define PCA9685_PRE_SCALE_DEFAULT 0x1E
+
+/* Helper macros PWM API period_count parameter */
+
+#define PCA9685_PWM_PERIOD_COUNT(update_rate, osc_clock) (osc_clock / update_rate)
+
+#define PCA9685_PWM_PERIOD_COUNT_INTERNAL(update_rate) \
+	PCA9685_PWM_PERIOD_COUNT(update_rate, PCA9685_OSC_CLOCK_INTERNAL)
+
+#define PCA9685_PWM_PERIOD_COUNT_PS(pre_scale) (PCA9685_PWM_STEPS * (pre_scale + 1))
+
+/**
+ * When using the internal oscillator this corresponds to an
+ * update rate of 1526 Hz
+ */
+#define PCA9685_PWM_PERIOD_COUNT_MIN PCA9685_PWM_PERIOD_COUNT_PS(PCA9685_PRE_SCALE_MIN)
+
+/**
+ * When using the internal oscillator this corresponds to an
+ * update rate of 24 Hz
+ */
+#define PCA9685_PWM_PERIOD_COUNT_MAX PCA9685_PWM_PERIOD_COUNT_PS(PCA9685_PRE_SCALE_MAX)
+
+/**
+ * When using the internal oscillator this corresponds to an
+ * update rate of 200 Hz
+ */
+#define PCA9685_PWM_PERIOD_COUNT_DEFAULT PCA9685_PWM_PERIOD_COUNT_PS(PCA9685_PRE_SCALE_DEFAULT)
```

Comment:
why is all that placed in a public header?

Reply:
The intent of these macros was to provide to application code the PCA9685 PWM period default and limits of the device.

Reply:
> The intent of these macros was to provide to application code the PCA9685 PWM period default and limits of the device.

That should be handled by the driver or driver API, this header should not be public.